test-framework/common/src/main/java/io/quarkus/test/common/WithTestResource.java

```diff
@@ -13,11 +13,20 @@
 /**
  * Used to define a test resource, which can affect various aspects of the application lifecycle.
  * <p>
+ * As of Quarkus 3.16, the default behavior of the annotation (meaning that {@code scope} has not been set)
+ * is that test classes annotated with same {@code WithTestResource} will <strong>not</strong> force a restart
+ * of Quarkus.
```

Comment:
Maybe I'm blind but while I see the code to order the restricted to class first, I'm not exactly sure how the tests with the same resources are actually grouped?

Reply:
Are you talking about `QuarkusTestProfileAwareClassOrderer`? I haven't changed that

Reply:
Yes I'm talking about this specific class. I think someone said it was ordering classes depending on resources and I don't see that in the code. I just see some grouping by if they have per-class test resources.

Don't we want to group classes which have similar test resources?

Reply:
I can look into this in a follow up (or maybe @famod can since he wrote the ordering code). 
For the time being, I would like to get this if we agree it's the proper functionality as the change has a very high chance of conflicting with any changes made in test handling code.

Reply:
> I don't see that in the code

At a first glimpse I don't see that either. :thinking: It is has been a while, so I'd need some time to look into it.
Until when do we need that adjustment (if needed after all)?

Reply:
Btw, `@QuarkusComponentTest` isn't covered either but I haven't looked into whether grouping such tests would make sense at all.

Reply:
>  It is has been a while, so I'd need some time to look into it. Until when do we need that adjustment (if needed after all)?

As this PR is breaking and meant to go into `3.16`, we still have a weeks

Reply:
> Btw, @QuarkusComponentTest isn't covered either but I haven't looked into whether grouping such tests would make sense at all.

Yeah, it probably makes sense to have those executed at the beginning or the end of the run, but that's definitely something for after this PR is in.

Reply:
> Btw, @QuarkusComponentTest isn't covered either but I haven't looked into whether grouping such tests would make sense at all.

Given they are supposed to be faster and lower level, let's execute them first.
But yes, definitely out of the scope for this PR.

Reply:
FYI, I haven't had the time to take a stab at the QuarkusTestProfileAwareClassOrderer change. I _might_ have some time for that by the end of next week.